ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -572,29 +572,15 @@ chainSyncClient mkPipelineDecision0 tracer cfg
           -- in two ways: 1. we adopted them, i.e., our chain changed (handled
           -- in this function); 2. we will /never/ adopt them, which is
           -- handled in the "no more intersection case".
-          case AF.splitAfterPoint theirFrag (AF.anchorPoint ourFrag') of
-           -- + Before the update to our fragment, both fragments were
-           --   anchored at the same anchor.
-           -- + We still have an intersection.
-           -- + The number of blocks after the intersection cannot have
-           --   shrunk, but could have increased.
-           -- + If it did increase, the anchor point will have shifted up.
-           -- + It can't have moved up past the intersection point (because
-           --   then there would be no intersection anymore).
-           -- + This means the new anchor point must be between the old anchor
-           --   point and the new intersection point.
-           -- + Since we know both the old anchor point and the new
-           --   intersection point exist on their fragment, the new anchor
-           --   point must also.
-           Nothing -> error
-               "anchor point must be on candidate fragment if they intersect"
-           Just (_, trimmedCandidateFrag) -> return $ Just $
+          case ourPre `AF.join` theirSuf of
```

Comment:
We talked about this in a call and came to the conclusion that whatever improvement of the best/average case this is, it is likely irrelevant for the decision of a user to downgrade their machine specs.

Concretely, a Cardano header has a maximum size of `1.1 kB`, and `ourPre` contains at most `k = 2160` headers, so we get a maximum size decrease of `1.1 kB × 2160 × 100 = 237.6 MB` very conservatively assuming 100 ChainSync clients, which we think is dwarfed by the overall memory usage by the node.